common/python/utility/jrpc_utility.py

```diff
@@ -15,6 +15,70 @@
 """
 jrpc_utility.py -- general jrpc helper function
 """
+import json
+from jsonschema import validate, ValidationError, SchemaError
+
+
+def request_validation(input_json_request):
+    """
+    Function to check basic json request parameters
+    Parameters:
+        -input_json_request: json to validate
+    Returns:
+        True and empty string on success and
+        False and string with error message on failure.
+    """
+    schema = {
+        "type": "object",
+        "uniqueItems": True,
+        "additionalProperties": False,
+        "properties": {
+            "params": {},
+            "jsonrpc": {
+                "type": "string",
+                "enum": ["2.0"],
+                "error_msg":
+                "Invalid data format or value for jsonrpc"},
+            "id": {
+                "type": "integer",
+                "error_msg":
+                "Invalid data format for id"},
+            "method": {
+                "type": "string",
+                "enum": [
+                    "WorkerLookUp",
+                    "WorkerLookUpNext",
+                    "WorkerRetrieve",
+                    "WorkOrderSubmit",
+                    "WorkOrderGetResult",
+                    "WorkOrderReceiptRetrieve"],
+                "error_msg":
+                "Invalid data format for method or method not supported"}
+        },
+        "required": ["jsonrpc", "id", "method", "params"]
+    }
+    try:
+        validate(input_json_request, schema)
+    except ValidationError as e:
+        if e.schema["error_msg"]:
+            raise ValueError(e.schema["error_msg"])
+        else:
+            raise ValueError(e.message)
+
+    return True, ""
+
+
+def json_duplicate_key_finder(input_json_str):
+    """Reject duplicate keys."""
+
+    input_json = json.loads(input_json_str)
+    input_string = json.dumps(input_json)
+
+    input_json_str = input_json_str.replace(" ", "")
+    input_string = input_string.replace(" ", "")
+
+    if len(input_string) < len(input_json_str):
```

Comment:
Brief explanation in comments of function doc will help follow the logic. Slightly skeptical of the logic here as well. Though not really able to point out when it can fail.

Reply:
Function did not work as expected hence Function has been removed.